server/wal/readonly_segments_group.go

```diff
@@ -301,6 +150,15 @@ func (r *readOnlySegmentsGroup) TrimSegments(offset int64) error {
 		return nil
 	}
 
+	removeIfExists := func(p string) error {
```

Comment:
Let us declare this function into some utils.  :)

Reply:
done, PTAL